extensions/amp-consent/0.1/consent-state-manager.js

```diff
@@ -14,76 +14,181 @@
  * limitations under the License.
  */
 
+import {Observable} from '../../../src/observable';
 import {Services} from '../../../src/services';
+import {dev} from '../../../src/log';
+
+const TAG = 'CONSENT-STATE-MANAGER';
+
+/**
+ * @enum {number}
+ */
+export const CONSENT_ITEM_STATE = {
+  UNKNOWN: 0,
+  GRANTED: 1,
+  REJECTED: 2,
+};
 
 export class ConsentStateManager {
   constructor(ampdoc) {
-    this.ampdoc = ampdoc;
+    /** @private {?../../../src/service/ampdoc-impl.AmpDoc} */
+    this.ampdoc_ = ampdoc;
 
+    /** @private {!Object<string, ConsentInstance>} */
     this.instances_ = {};
+
+    /** @private {!Object<string, ?Observable>}*/
+    this.consentChangeObservable_ = {};
+
+    /** @private {!Object<string, ?function()>} */
+    this.consentReadyResolver_ = {};
```

Comment:
same question as above, since these are collections should they be plural?

Reply:
changed.